template_content/{% if use_typescript_jest %}tests{% endif %}/{% if use_typescript_jest %}{{ contract_name.split('_')|join('-') }}.spec.ts{% endif %}.jinja

```diff
@@ -0,0 +1,48 @@
+import { algorandFixture } from '@algorandfoundation/algokit-utils/testing'
+import { {{ contract_name.split('_')|map('capitalize')|join }}Client } from '../smart_contracts/artifacts/{{ contract_name }}/client'
+import { Account, Algodv2, Indexer } from 'algosdk'
+
+describe('{{ contract_name.split('_')|join(' ') }} contract', () => {
+  const localnet = algorandFixture()
+  beforeEach(localnet.beforeEach)
+
+  const deploy = async (account: Account, algod: Algodv2, indexer: Indexer) => {
+    const client = new {{ contract_name.split('_')|map('capitalize')|join }}Client(
+      {
+        resolveBy: 'creatorAndName',
+        findExistingUsing: indexer,
+        sender: account,
+        creatorAddress: account.addr,
+      },
+      algod,
+    )
+    await client.deploy({
+      allowDelete: true,
+      allowUpdate: true,
+      onSchemaBreak: 'replace',
+      onUpdate: 'update',
+    })
+    return { client }
+  }
+
+  test('says hello', async () => {
+    const { algod, indexer, testAccount } = localnet.context
+    const { client } = await deploy(testAccount, algod, indexer)
+
+    const result = await client.hello({ name: 'World' })
+
+    expect(result.return).toBe('Hello, World')
+  })
+
+  test('simulate says hello with correct budget consumed', async () => {
+    const { algod, indexer, testAccount } = localnet.context
+    const { client } = await deploy(testAccount, algod, indexer)
+    const atc = await client.compose().hello({ name: 'World' }).hello({ name: 'Jane' }).atc()
```

Comment:
Should we expose simulate from the app client compose?

Reply:
Adding support for simulate in the generators
https://github.com/algorandfoundation/algokit-client-generator-py/pull/17
https://github.com/algorandfoundation/algokit-client-generator-ts/pull/86

Once that is merged, I'll update this PR.